### DISCUSSION TITLE
Feature: Allow optional prefixes in App directory to avoid repetitive "page.jsx" filenames

### DISCUSSION THREAD

**Discussion by RobJacobson** [vercel/next.js#49725] [Ideas]
### Goals

Provide a better developer experience by allowing more than repetitive "page.jsx" filenames in the App directory.

### Non-Goals

This would not change the current behavior of file-system routing in the App directory, other than allowing optional prefixes as a convenience for developers.

### Background

I'm new to Next 13. I appreciate the new App Directory features. However, I find it confusing to work with the naming requirements in VS Code. 

If I'm working on multiple pages, all of the tabs in VS Code say "page.tsx," as do all of the files in the file explorer. When I'm confused (which happens easily and often!), I need to look at the folder name, which is counterintuitive. 

To quote a random user on Reddit (an actual comment):

"I love next but god do I hate page.tsx page.tsx  page.tsx  page.tsx  page.tsx page.tsx page.tsx page.tsx page.tsx page.tsx page.tsx page.tsx page.tsx page.tsx."

To quote a response: "Would love to be able to prefix like index.page.tsx, dashboard.page.txs, etc."

(Credit where credit is due!)

### Proposal

Please consider allowing _optional_ prefixes for `page.jsx` when using the App Directory. E.g., a developer could name their pages files like this:

`app/products/products.page.jsx`
`app/users/users.page.jsx`

This system could be extended for other common files, like `layout.jsx`. 

This would be completely optional. Developers could continue to use just `page.jsx` (or `.tsx`) if desired. The prefix would have no semantic meaning, but would just be a convenience for developer who prefer this naming convention.

This could be implemented by having the compiler simply ignore any prefixes. E.g., it would treat `products.page.jsx` as completely equivalent to `page.jsx`. 

There is a risk that users might attempt to place multiple "page.jsx" in a single App Directory folder, as with the traditional Pages Directory. E.g., they might mistakenly do something like:

`app/products/productone.page.jsx`
`app/products/producttwo.page.jsx`

This risk could be mitigated by either of the following:

1. Requiring the prefix (if any) to match the name of the last directory segment, as above. E.g., either of the following would be allowed, but not both:

`app/products/products.page.jsx`

`app/products/page.jsx`

The compiler would treat any other filenames as an error.

2. Allow any prefixes, while the compiler ignores them. The compiler would check for multiple `... .pages.jsx` files, and throw an error if multiple files exist.

Thanks for considering this suggestion.

**Comment by timdk** [comment]
Related discussion: https://github.com/vercel/next.js/discussions/49645
